### PR TITLE
Type hint check and corrections

### PR DESCRIPTION
**Pull request by oncleben31**
Add type hint checks with mypy and typeguard following best practices of hypermodern_python (see #18).
Code update was needed to correct all errors detected by mypy.

### CODE REVIEW

File: src/meteofrance/helpers.py [hunk 1]
```diff
@@ -2,7 +2,9 @@
 """Météo-France helpers."""
 import math
 from datetime import datetime
+from typing import Dict
 from typing import List
+from typing import Optional
 from typing import Tuple
```
Comment (Quentame) [inline src/meteofrance/helpers.py:8]: I would prefer one line import, same bellow.

File: src/meteofrance/helpers.py [hunk 2]
```diff
@@ -27,7 +31,9 @@ def get_warning_text_status_from_indice_color(int_color: int, lang: str = "fr")
         return ALERT_COLOR_LIST_EN[int_color]
```
Comment (Quentame) [inline src/meteofrance/helpers.py:31]: remove unnecessary else

File: src/meteofrance/helpers.py [hunk 3]
```diff
@@ -16,7 +18,9 @@
 from .model.place import Place
 
 
-def get_warning_text_status_from_indice_color(int_color: int, lang: str = "fr") -> str:
+def get_warning_text_status_from_indice_color(
+    int_color: int, lang: str = "fr"
+) -> Optional[str]:
```
Comment (Quentame) [inline src/meteofrance/helpers.py:23]: remove optional

File: src/meteofrance/helpers.py [hunk 4]
```diff
@@ -27,7 +31,9 @@ def get_warning_text_status_from_indice_color(int_color: int, lang: str = "fr")
         return ALERT_COLOR_LIST_EN[int_color]
 
 
-def get_phenomenon_name_from_indice(int_phenomenon: int, lang: str = "fr") -> str:
+def get_phenomenon_name_from_indice(
+    int_phenomenon: int, lang: str = "fr"
+) -> Optional[str]:
```
Comment (Quentame) [inline src/meteofrance/helpers.py:36]: same here for else and optional

File: src/meteofrance/helpers.py [hunk 5]
```diff
@@ -46,7 +52,9 @@ def is_coastal_department(department_number: str) -> bool:
     return result
 
 
-def readeable_phenomenoms_dict(list_phenomenoms: list, language: str = "fr") -> dict:
+def readeable_phenomenoms_dict(
+    list_phenomenoms: List[Dict[str, int]], language: str = "fr"
+) -> Dict[Optional[str], Optional[str]]:
```
Comment (Quentame) [inline src/meteofrance/helpers.py:57]: so without optional

File: src/meteofrance/session.py [hunk 6]
```diff
@@ -52,6 +56,6 @@ class MeteoNetSession(MeteoFranceSession):
     host: str = METEONET_API_URL
 
     # TODO: convert to class method
```
Comment (Quentame) [inline src/meteofrance/session.py:58]: We can remove those TODO

File: src/meteofrance/helpers.py (hunk 1 above)
Comment (oncleben31) [reply src/meteofrance/helpers.py:8]: I use [reorder-python-imports](https://pypi.org/project/reorder-python-imports/) to check and order automatically the import. They split imports to reduce merge conflicts.
I want to continue this way.

File: src/meteofrance/helpers.py (hunk 2 above)
Comment (oncleben31) [reply src/meteofrance/helpers.py:31]: OK fixed

File: src/meteofrance/helpers.py (hunk 3 above)
Comment (oncleben31) [reply src/meteofrance/helpers.py:23]: Currently optional is needed due to the fact that first element of ALERT_COLOR_LIST_FR and ALERT_COLOR_LIST_EN is None.
Another way could be to check if 0 < int_color < 5 and raise a ValueError or IndexError exception if False.

File: src/meteofrance/helpers.py (hunk 4 above)
Comment (oncleben31) [reply src/meteofrance/helpers.py:36]: else fixed.
Optional same as previous discussion.

File: src/meteofrance/helpers.py (hunk 5 above)
Comment (oncleben31) [reply src/meteofrance/helpers.py:57]: Linked to previous discussion.

File: src/meteofrance/session.py (hunk 6 above)
Comment (oncleben31) [reply src/meteofrance/session.py:58]: OK

**Comment by oncleben31** [comment]
Don't merge yet, I need to correct the GitHub workflow about code coverage.

**Comment by oncleben31** [comment]
Code coverage check in GA is OK now. We can merge as soon as the code review is done.

**Pull request merged**